staging/src/k8s.io/client-go/tools/record/event.go

```diff
@@ -338,11 +338,14 @@ func (recorder *recorderImpl) generateEvent(object runtime.Object, annotations m
 	event := recorder.makeEvent(ref, annotations, eventtype, reason, message)
 	event.Source = recorder.source
 
-	go func() {
-		// NOTE: events should be a non-blocking operation
```

Comment:
for backporting, would it be less invasive to switch this to something like this:

```go
defer func(){
  if err := recover(); err != nil {
    utilruntime.HandleError(err)
  }
}()
```

I'd like more soak time on the doubling of the event queue before backporting

Reply:
cc @DirectXMan12